src/ContainerCommitModal.jsx

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import {Modal, Button} from 'patternfly-react';
+import cockpit from 'cockpit';
+
+const _ = cockpit.gettext;
+
+class ContainerCommitModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.initialState = {
+            imageName: "",
+            tag: "",
+            author:"",
+            message: "",
+            command: this.props.containerWillCommit.Config ? this.props.containerWillCommit.Config.Cmd.join(" ") : "",
+            pause: true,
+            setonbuild: false,
+            onbuild: [""],
+            format: "oci",
+            selectedFormat: "oci",
+            onbuildDisabled: true,
+        };
+        this.state = {...this.initialState};
```

Comment:
Do you really need the initialState variable?
I see you use it in the cancel handler to reset the components state, but this would have the same results as just keeping this object in the state variable, which will get reinitialized next time the component will re-render. Or I am missing something?

Reply:
The initialState is used for changing the checkbox or input field back to default when to reopen the same modal. If the first time I open the modal and make some changes to checkbox or input filed and closed the modal using cancel. Then I use the commit button to open the modal again, the modal should go back to the default.